### PR TITLE
Added Binary Search [CPP]

### PR DESCRIPTION
**Pull request by ManBearPigg**
Fixes #{ISSUE_NUMBER}

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

**Comment by singhpratyush** [comment]
Hi! Thanks for showing interest in contributing to this repository.

Please make sure that you have ticked the points mentioned in PR description correctly.

Thanks again!

**Comment by ManBearPigg** [comment]
Hello,

My pull request is failing a check for "trailing whitespace". I know that binary_search.cpp contains whitespace, and I know that trailing means "at the end of", but I still don't understand where the problematic trailing whitespace is in my program. Would anyone familiar with this check system kindly point out where the offending "trailing whitespace" is? 

Thanks.
manbearpigg

**Comment by singhpratyush** [comment]
```
The command "coala --ci" exited with 0.
1.06s$ eclint check "**/*.{java,c,cpp,js,cs}"
binary_search/binary_search.cpp
    [09:03] ❌️ unexpected trailing whitespace (EditorConfig trim_trailing_whitespace https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#trim_trailing_whitespace)
        09|// 
    1bc32789 (manbearpigg <sam-allick@hotmail.ca> 10/2/2017, 4:35:04 PM)
```

Looks like it is at line 9,

**Comment by singhpratyush** [comment]
Also, please make corrections according to this commit - e654f6411e690e859ccee0fbe999123be88f5107.

File: binary_search/binary_search.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#include <iostream>
+#include <vector>
+using namespace std;
+
+//------------------------------------------------
```
Comment (singhpratyush) [inline binary_search/binary_search.cpp]: Please make the comments in this style - https://github.com/iiitv/algos/blob/master/longest_palindromic_substring/longest_palindromic_substring.cpp.

File: binary_search/binary_search.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+// int target: element to search for.
+//
+// OUTPUT
+// int: index of target if it exists, else -1.
+//------------------------------------------------
+int binary_search(vector<int> list, int target) {
+	int first = 0;
+	int last = list.size() - 1;
+	while (first <= last) {
+		int mid = (first + last) / 2;
```
Comment (singhpratyush) [inline binary_search/binary_search.cpp]: https://github.com/iiitv/algos/blob/master/binary_search/binary_search.c#L19

**Comment by aashutoshrathi** [comment]
Please Create an Issue for Binary Search first and add issue number to issue description, also read Contribution Guidelines.

**Comment by ManBearPigg** [comment]
I'm abandoning this pull request because you add way too much overhead and regulations into what should be a simple repo.

**Comment by singhpratyush** [comment]
The issues are needed so that we can ensure that no two people are putting in their time for the same thing. Hope you understand that keeping with the guidelines is important and opening an issue before submitting a patch is nothing new or unnecessary here.

Feel free to reopen if you consider working further with this PR. Thanks.

**Pull request closed without merging**